### PR TITLE
Remove deprecated code

### PR DESCRIPTION
**Pull request by joaander**
## Description

Remove deprecated uses of BoxResize and rewrite Kern-Frenkel tutorial with builtin classes.

The deprecated use of HPMC walls remains as there is no alternative at this time.

Also add a `jupytext` configuration so that authors can edit prose in a Markdown editor of their choice. Authors should not commit the synced md files at this time.

## Checklist:
- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-examples/blob/trunk-minor/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-examples/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the [list of authors](https://github.com/glotzerlab/hoomd-examples/blob/trunk-minor/AUTHORS.md).

### CODE REVIEW

**Comment by joaander** [comment]
Note: This PR also fixes a `ruff-format` configuration and therefore there are formatting changes in unrelated files.

I also fixed several typos in unrelated files as I found them.

File: 06-Modelling-Rigid-Bodies/01-Introduction-to-Rigid-Bodies.ipynb [hunk 1]
```diff
@@ -254,11 +254,11 @@
        "  <rdf:RDF xmlns:dc=\"http://purl.org/dc/elements/1.1/\" xmlns:cc=\"http://creativecommons.org/ns#\" xmlns:rdf=\"http://www.w3.org/1999/02/22-rdf-syntax-ns#\">\n",
```
Comment (tcmoore3) [inline 06-Modelling-Rigid-Bodies/01-Introduction-to-Rigid-Bodies.ipynb:254]: <p>change</p><blockquote>place the central particle of each body first and set the <code>body</code> property of all members to the index&nbsp;of the constituent particle.</blockquote><p>to</p><blockquote>place the central particle of each body first and set the <code>body</code> property of all member particles (central <em>and</em> constituent) to the index&nbsp;of the central particle.</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/hoomd-examples/pull/131/discussion/#comment-1673017865'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='53'/>

File: 07-Modelling-Patchy-Particles/01-Kern-Frenkel-Model.ipynb [hunk 2]
```diff
@@ -5,7 +5,7 @@
    "id": "0",
```
Comment (tcmoore3) [inline 07-Modelling-Patchy-Particles/01-Kern-Frenkel-Model.ipynb:5]: <p>Change</p><blockquote>Some systems are better represented by anisotropic pair potentials that depend on particle relative orientations. </blockquote><p>to</p><blockquote>Some systems are better represented by anisotropic pair potentials that depend on relative particle orientations. </blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/hoomd-examples/pull/131/discussion/#comment-1673017868'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='1'/>

File: 07-Modelling-Patchy-Particles/02-Simulating-a-System-of-Patchy-Particles-with-HPMC.ipynb [hunk 3]
```diff
@@ -66,7 +66,7 @@
    "metadata": {},
```
Comment (tcmoore3) [inline 07-Modelling-Patchy-Particles/02-Simulating-a-System-of-Patchy-Particles-with-HPMC.ipynb:66]: <p>Replace</p><blockquote>As discussed in the Logging tutorial,</blockquote><p>with </p><blockquote>As discussed in the [Logging tutorial](../02-Logging/02-Saving-Array-Quantities.html),</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/hoomd-examples/pull/131/discussion/#comment-1673017870'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='25'/>

**Review by tcmoore3 — APPROVED** [review]

**Pull request merged**